subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv_pdu.h

```diff
@@ -101,7 +101,8 @@ static inline void lll_adv_aux_data_enqueue(struct lll_adv_aux *lll,
 	lll_adv_pdu_enqueue(&lll->data, idx);
 }
 
-static inline struct pdu_adv *lll_adv_aux_data_peek(struct lll_adv_aux *lll)
+static inline struct pdu_adv *
+	lll_adv_aux_data_peek(const struct lll_adv_aux *const lll)
 {
 	return (void *)lll->data.pdu[lll->data.last];
 }
```

Comment:
Isn't using `const` here misleading, as we are returning a non-const pointer from the struct?

Reply:
I am not sure I followed that at all. 

The issue here is that we expose the function with a `const` parameter, which typically means that the caller can be sure that the parameter's values are not changed (of course, with a cast to `(void *)` we can always do what we want in C). 

Since we are returning a field from the struct, as non-const, the `const` argument qualifier not only has no meaning (as we can actually change a field in the `const` parameter afterwards), but is misleading, and serves no purpose.

Reply:
@Thalley is right. You just `un-const` part of a structure inside the function and return it as if it were never const. That can break something in a nasty way. This is actually breaking a chain objects that are `const` or are pointed by pointers to `const`.

Reply:
Allow me to present an sample example of why I think this is an issue:

```c
#include <stdio.h>

struct my_struct {
	int a;
};

static int *foo(const struct my_struct *const ms)
{
	return (void *)&ms->a;
}

static void bar(const struct my_struct *const ms)
{
	int *a = foo(ms);

	/* We can now modify a */
	*a = 15;
}

int main()
{
	struct my_struct ms = { .a = 10 };

	printf("a: %d\n", ms.a); /* prints 10 */

	/* Calling bar expecting no change since argument is const */
	bar(&ms);

	printf("a: %d\n", ms.a); /* prints 15 */

	return 0;
}
```

If you need to cast it to a non-const pointer in other to build it, then it sounds like the callers of this function can also not use `const` as the content of the `pdu` (a field of the `struct lll_adv_aux`) can be/is changed.

Reply:
What `lll->data.pdu[lll->data.last]` points to is never a `const`, it is a pointer that is allocated with PDU buffers from a pool and PDU contents are always modified. The pointer stored is qualified as constant by the change in this PR, but the contents are not constant.

https://github.com/zephyrproject-rtos/zephyr/blob/55be9786bcf0c5b3ba4e04fe8f4df8f0f4e8febc/subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv_types.h#L15-L18

Reply:
You are absolutely correct Vinayak :) Please excuse my lack of knowledge of the LLL data structures (I expected the `pdu_adv` to be part of the `lll_adv_aux`, and not just be a reference kept in the `data` array). 

This should be fine